packages/@aws-cdk/aws-sagemaker/lib/model.ts

```diff
@@ -0,0 +1,405 @@
+   * SageMaker Model to connect to network targets.
+   *
+   * Only used if 'vpc' is supplied.
+   *
+   * @default true
+   */
+  readonly allowAllOutbound?: boolean;
+}
+
+/**
+ * Defines a SageMaker Model.
+ */
+export class Model extends ModelBase {
+  /**
+   * Imports a Model defined either outside the CDK or in a different CDK stack.
+   * @param scope the Construct scope.
+   * @param id the resource id.
+   * @param modelName the name of the model.
+   */
+  public static fromModelName(scope: Construct, id: string, modelName: string): IModel {
+    return Model.fromModelAttributes(scope, id, { modelName });
+  }
+
+  /**
+   * Imports a Model defined either outside the CDK or in a different CDK stack.
+   * @param scope the Construct scope.
+   * @param id the resource id.
+   * @param attrs the attributes of the model to import.
+   */
+  public static fromModelAttributes(scope: Construct, id: string, attrs: ModelAttributes): IModel {
+    const modelName = attrs.modelName;
+    const role = attrs.role;
+
+    class Import extends ModelBase {
+      public readonly modelName = modelName;
+      public readonly role = role;
+      public readonly grantPrincipal: iam.IPrincipal;
+      public readonly modelArn: string;
+
+      constructor(s: Construct, i: string) {
+        super(s, i);
+
+        this.modelArn = cdk.Stack.of(this).formatArn({
+          service: 'sagemaker',
+          resource: 'model',
+          resourceName: this.modelName,
+        });
+        this.grantPrincipal = role || new iam.UnknownPrincipal({ resource: this });
+        if (attrs.securityGroups) {
+          this._connections = new ec2.Connections({
+            securityGroups: attrs.securityGroups,
+          });
+        }
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Returns the ARN of this model.
+   * @attribute
+   */
+  public readonly modelArn: string;
+  /**
+   * Returns the name of the model.
+   * @attribute
+   */
+  public readonly modelName: string;
+  /**
+   * Execution role for SageMaker Model
+   */
+  public readonly role?: iam.IRole;
+  /**
+   * The principal this Model is running as
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+  private readonly subnets: ec2.SelectedSubnets | undefined;
+  private readonly containers: CfnModel.ContainerDefinitionProperty[] = [];
+
+  constructor(scope: Construct, id: string, props: ModelProps = {}) {
+    super(scope, id, {
+      physicalName: props.modelName,
+    });
+
+    this._connections = this.configureNetworking(props);
+    this.subnets = (props.vpc) ? props.vpc.selectSubnets(props.vpcSubnets) : undefined;
+
+    // set the sagemaker role or create new one
+    this.role = props.role || this.createSageMakerRole();
+    this.grantPrincipal = this.role;
+
+    (props.containers || []).map(c => this.addContainer(c));
+
+    // apply a name tag to the model resource
+    cdk.Tags.of(this).add(NAME_TAG, this.node.path);
```

Comment:
maybe this is just my unfamiliarity with the resource, but is there a reason why we always add a tag to the resource?

Reply:
I actually thought this was a more prevalent pattern in the CDK itself, but upon [performing a more recent search](https://github.com/aws/aws-cdk/search?q=Tags.of%28this%29+this.node.path&type=code), it looks like only a handful of modules do this. I'm not personally attached to this tag, so if you'd like, I can just rip this out.

Reply:
I guess I've never seen this before, which is the cause for my confusion. In these scenarios I prefer to leave them out. We can always add them in later, but we can't remove it later if it's already there.